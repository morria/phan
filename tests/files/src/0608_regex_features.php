<?php
call_user_func(function () {
    if (preg_match('/(?:a|ba)(c+)/', 'acc', $matches)) {
        var_export($matches);
        echo $matches[2];
    }
    $matches = [];
    if (preg_match('/(?:a|ba)(?# comments)(c+)/', 'acc', $matches)) {
        var_export($matches);
        echo $matches[2];
    }
    $matches = [];
    if (preg_match('/(?:a|ba)(?P<name>c+)/', 'acc', $matches)) {
        var_export($matches);
        echo $matches[2];
    }
    echo "Looking for match\n";
    $matches = [];
    if (preg_match('/(?:a|ba)(?\'a\'c+)/', 'acc', $matches)) {
        var_export($matches);
        echo $matches[2];
    }
    if (preg_match('/(?:a|ba)(?<de>c+)/', 'acc', $matches)) {
        var_export($matches);
        echo $matches[2];
    }
    if (preg_match('/(?i)(?:a|ba)(?-i)(?<caseInsensitive>c+)/', 'Acc', $matches)) {
        var_export($matches);
        echo $matches[2];
    }
});
