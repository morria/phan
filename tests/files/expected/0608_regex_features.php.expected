%s:5 PhanTypeInvalidDimOffset Invalid offset 2 of array type array{0:string,1:string}
%s:5 PhanTypeSuspiciousEcho Suspicious argument null for an echo/print statement
%s:10 PhanTypeInvalidDimOffset Invalid offset 2 of array type array{0:string,1:string}
%s:10 PhanTypeSuspiciousEcho Suspicious argument null for an echo/print statement
%s:15 PhanTypeInvalidDimOffset Invalid offset 2 of array type array{0:string,name:string,1:string}
%s:15 PhanTypeSuspiciousEcho Suspicious argument null for an echo/print statement
%s:21 PhanTypeInvalidDimOffset Invalid offset 2 of array type array{0:string,a:string,1:string}
%s:21 PhanTypeSuspiciousEcho Suspicious argument null for an echo/print statement
%s:25 PhanTypeInvalidDimOffset Invalid offset 2 of array type array{0:string,de:string,1:string}
%s:25 PhanTypeSuspiciousEcho Suspicious argument null for an echo/print statement
%s:29 PhanTypeInvalidDimOffset Invalid offset 2 of array type array{0:string,caseInsensitive:string,1:string}
%s:29 PhanTypeSuspiciousEcho Suspicious argument null for an echo/print statement
